### PR TITLE
Update Harmony and MonoMod to support .NET 8

### LINKED ISSUE #434: .Net 8 Support xUnit - Harmony -> MonoMod Dependency
[//]: # (
. Note: for support questions, please use Stackoverflow or Gitter**. 
. This repository's issues are reserved for feature requests and bug reports.
.
. In case of any problems with Allure Jenkins plugin** please use the following repository 
. to create an issue: https://github.com/jenkinsci/allure-plugin/issues
.
. Make sure you have a clear name for your issue. The name should start with a capital 
. letter and no dot is required in the end of the sentence. An example of good issue names:
.
. - The report is broken in IE11
. - Add an ability to disable default plugins
. - Support emoji in test descriptions
)

#### I'm submitting a ... 
  - [x] bug report
  - [ ] feature request
  - [ ] support request => Please do not submit support request here, see note at the top of this template.

#### What is the current behavior?
Running dotnet test will result in an error:
<pre>
[xUnit.net 00:00:00.20] Unable to patch XunitTestRunner's Void .ctor(Xunit.Abstractions.ITest, Xunit.Sdk.IMessageBus, System.Type, System.Object[], System.Reflection.MethodInfo, System.Object[], System.String, System.Collections.Generic.IReadOnlyList`1[Xunit.Sdk.BeforeAfterTestAttribute], Xunit.Sdk.ExceptionAggregator, System.Threading.CancellationTokenS
ource): System.TypeInitializationException: The type initializer for 'MonoMod.Utils.DMDEmitDynamicMethodGenerator' threw an exception.
 ---> System.InvalidOperationException: Cannot find returnType fieeld on DynamicMethod
   at MonoMod.Utils.DMDEmitDynamicMethodGenerator..cctor()
   --- End of inner exception stack trace ---
   at MonoMod.Utils.DMDEmitDynamicMethodGenerator.GenerateCore(DynamicMethodDefinition dmd, Object context)
   at MonoMod.Utils.DMDGenerator`1.Generate(DynamicMethodDefinition dmd, Object context)
   at MonoMod.Utils.DynamicMethodDefinition.Generate(Object context)
   at MonoMod.Utils.DynamicMethodDefinition.Generate()
   at HarmonyLib.MethodPatcher.CreateReplacement(Dictionary`2& finalInstructions)
   at HarmonyLib.PatchFunctions.UpdateWrapper(MethodBase original, PatchInfo patchInfo)
   at HarmonyLib.PatchProcessor.Patch()
   at HarmonyLib.Harmony.Patch(MethodBase original, HarmonyMethod prefix, HarmonyMethod postfix, HarmonyMethod transpiler, HarmonyMethod finalizer)
   at Allure.XUnit.AllureXunitPatcher.PatchXunitTestRunnerCtors(Harmony patcher)
[xUnit.net 00:00:00.20] No constructors of XunitTestRunner were pathched. Some theories may miss their parameters in the report
</pre>

#### If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem
Run dotnet test with an xUnit project in .net 8

#### What is the expected behavior?
Report is generated

#### What is the motivation / use case for changing the behavior?
Support .Net 8, make others aware that base dependency needs to be updated to support it.

#### Please tell us about your environment:

- Allure version:        2.10.0
- Test framework:        xunit@2.6.2
- Allure adaptor:        allure-xunit@2.10.0
- Generate report using: dotnet test for a .Net 8 project

#### Other information 
I cloned the project, started updating dependencies, testing with .net 8, and have found this error:
<pre>
OneTimeSetUp: System.TypeInitializationException : The type initializer for 'MonoMod.Utils.DMDEmitDynamicMethodGenerator' threw an exception.
  ----> System.InvalidOperationException : Cannot find returnType fieeld on DynamicMethod
   at MonoMod.Utils.DMDEmitDynamicMethodGenerator.GenerateCore(DynamicMethodDefinition dmd, Object context)
   at MonoMod.Utils.DMDGenerator`1.Generate(DynamicMethodDefinition dmd, Object context)
   at MonoMod.Utils.DynamicMethodDefinition.Generate(Object context)
   at MonoMod.Utils.DynamicMethodDefinition.Generate()
   at HarmonyLib.MethodPatcher.CreateReplacement(Dictionary`2& finalInstructions)
   at HarmonyLib.PatchFunctions.UpdateWrapper(MethodBase original, PatchInfo patchInfo)
   at HarmonyLib.PatchProcessor.Patch()
   at HarmonyLib.Harmony.Patch(MethodBase original, HarmonyMethod prefix, HarmonyMethod postfix, HarmonyMethod transpiler, HarmonyMethod finalizer)
   at Allure.SpecFlowPlugin.SelectiveRun.AllureSpecFlowPatcher.PatchRunnerFactory(Harmony patcher, MethodBase factoryCandidate) in C:\projects\other\allure-csharp\Allure.SpecFlowPlugin\SelectiveRun\AllureSpecFlowPatcher.cs:line 84
   at Allure.SpecFlowPlugin.SelectiveRun.AllureSpecFlowPatcher.PatchRunnerFactories(Harmony patcher, IEnumerable`1 factoryCandidates) in C:\projects\other\allure-csharp\Allure.SpecFlowPlugin\SelectiveRun\AllureSpecFlowPatcher.cs:line 75
   at Allure.SpecFlowPlugin.SelectiveRun.AllureSpecFlowPatcher.InjectTestPlanCheckToTestRunner(Harmony patcher) in C:\projects\other\allure-csharp\Allure.SpecFlowPlugin\SelectiveRun\AllureSpecFlowPatcher.cs:line 38
   at Allure.SpecFlowPlugin.SelectiveRun.AllureSpecFlowPatcher.InjectTestPlanSupport(IUnitTestRuntimeProvider unitTestRuntimeProvider) in C:\projects\other\allure-csharp\Allure.SpecFlowPlugin\SelectiveRun\AllureSpecFlowPatcher.cs:line 33
   at Allure.SpecFlowPlugin.SelectiveRun.AllureSpecFlowPatcher.EnsureTestPlanSupportInjected(IUnitTestRuntimeProvider unitTestRuntimeProvider) in C:\projects\other\allure-csharp\Allure.SpecFlowPlugin\SelectiveRun\AllureSpecFlowPatcher.cs:line 24
   at Allure.SpecFlowPlugin.AllureBindingInvoker..ctor(SpecFlowConfiguration specFlowConfiguration, IErrorProvider errorProvider, ISynchronousBindingDelegateInvoker synchronousBindingDelegateInvoker, ITestRunnerManager testRunnerManager, IUnitTestRuntimeProvider unitTestRuntimeProvider) in C:\projects\other\allure-csharp\Allure.SpecFlowPlugin\AllureBindingInvoker.cs:line 45
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithManyArgs(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
--InvalidOperationException
   at MonoMod.Utils.DMDEmitDynamicMethodGenerator..cctor()
</pre>

Which points to this code:
https://github.com/allure-framework/allure-csharp/blob/51b6f159e24aefef3a0b85a6dc5895c41a10365c/Allure.SpecFlowPlugin/SelectiveRun/AllureSpecFlowPatcher.cs#L84

And I'd guess then that this applies:
https://github.com/MonoMod/MonoMod/issues/149

.Net 8 support PR's are:
https://github.com/MonoMod/MonoMod/pull/150

Hoping this saves others time, just need to wait for MonoMod .Net 8 support it seems, which has an open PR for it.

### PR DESCRIPTION
**Pull request by TonEnfer**
#### Checklist
- [x] [Sign Allure CLA][cla]
- [ ] Provide unit tests

[cla]: https://cla-assistant.io/accept/allure-framework/allure2

### CODE REVIEW

File: Allure.SpecFlowPlugin/Allure.SpecFlowPlugin.csproj [hunk 1]
```diff
@@ -26,7 +26,8 @@
 
     <ItemGroup>
         <PackageReference Include="CsvHelper" Version="30.0.1" />
-        <PackageReference Include="Lib.Harmony" Version="2.3.0-prerelease.2" />
+        <PackageReference Include="Lib.Harmony" Version="2.3.0-prerelease.5" />
+        <PackageReference Include="MonoMod.Utils" Version="25.0.4" />
```
Comment (delatrie) [inline Allure.SpecFlowPlugin/Allure.SpecFlowPlugin.csproj:30]: Harmony 2.3.0 already includes MonoMod.Utils v25.0.4+. We can remove this line now. Same for xUnit

**Review by delatrie — CHANGES_REQUESTED** [review]
Hi, @TonEnfer !
Thank you for the contribution!

Let's update straight to Harmony 2.3.0 (it was released a couple of days ago).

**Review by delatrie — APPROVED** [review]

**Comment by AlexBar** [comment]
Could you guys update nuget package to include these changes?

**Comment by delatrie** [comment]
@AlexBar Sorry for the delay. Version 2.12.0 that contains the fix has just been released!

**Pull request merged**